cc65-master, cc65 compiler crashes.

Running Linux Mint 20.1, compiling Fuzix latest with cc65 latest gives the following:
cl65 -c -t none -O -D__STDC__ -c -O -I/home/adam/work/FUZIX/Library/include -I/home/adam/work/FUZIX/Library/include/6502 stak.c
cl65 -c -t none -O -D__STDC__ -c -O -I/home/adam/work/FUZIX/Library/include -I/home/adam/work/FUZIX/Library/include/6502 string.c
cl65 -c -t none -O -D__STDC__ -c -O -I/home/adam/work/FUZIX/Library/include -I/home/adam/work/FUZIX/Library/include/6502 word.c
cl65 -c -t none -O -D__STDC__ -c -O -I/home/adam/work/FUZIX/Library/include -I/home/adam/work/FUZIX/Library/include/6502 xec.c
cl65: Subprocess 'cc65' aborted by signal 11
make[2]: *** [Makefile.6502:26: xec.o] Error 1
make[2]: Leaving directory '/home/adam/work/FUZIX/Applications/V7/cmd/sh'
make[1]: *** [Makefile:16: sh] Error 2
make[1]: Leaving directory '/home/adam/work/FUZIX/Applications'
make: *** [Makefile:99: apps] Error 2

Running with cc65 release V2.19 works correctly.

More details in https://github.com/EtchedPixels/FUZIX/issues/862

Just tried cc65-head with Fuzix-head and it does not crash anymore, BUT it does not work correctly.
Using cc65-2.19 completes "make all" without any problem.
Using cc65_head fails and shows this at the end, which I interpret as the new version of cc65 still not liking xec.c for some reason:

cl65 -c -t none -O -D__STDC__ -c -O -I/home/adam/work/PZ1-laptop/FUZIX_head/Library/include -I/home/adam/work/PZ1-laptop/FUZIX_head/Library/include/6502  xec.c
cl65: Subprocess 'cc65' aborted by signal 11
make[2]: *** [Makefile.6502:26: xec.o] Error 1
make[2]: Leaving directory '/home/adam/work/PZ1-laptop/FUZIX_head/Applications/V7/cmd/sh'
make[1]: *** [Makefile:16: sh] Error 2
make[1]: Leaving directory '/home/adam/work/PZ1-laptop/FUZIX_head/Applications'
make: *** [Makefile:103: apps] Error 2

--- a/src/cc65/coptstop.c
+++ b/src/cc65/coptstop.c
@@ -1113,9 +1113,9 @@ static unsigned Opt_a_toscmpbool (StackOpData* D, const char* BoolTransformer)
 
     D->IP = D->OpIndex + 1;
 
-    if (!D->RhsMultiChg &&
-        (D->Rhs.A.LoadEntry->Flags & CEF_DONT_REMOVE) == 0 &&
-        (D->Rhs.A.Flags & LI_DIRECT) != 0) {
+    if (!D->RhsMultiChg                     &&
+        (D->Rhs.A.Flags & LI_DIRECT) != 0   &&
+        (D->Rhs.A.LoadEntry->Flags & CEF_DONT_REMOVE) == 0) {
 
         /* cmp */
         AddOpLow (D, OP65_CMP, &D->Rhs);

